wrench/src/parse_function.rs

```diff
@@ -0,0 +1,90 @@
+use std::str::CharIndices;
+// A crapy parser for parsing strings like "translate(1, 3)"
+
+pub fn parse_function(s: &str) -> (&str, Vec<&str>) {
+    // XXX: This it not particular easy to read. Sorry.
+    struct Parser<'a> {
+        itr: CharIndices<'a>,
+        start: usize,
+        o: Option<<CharIndices<'a> as Iterator>::Item>,
+    }
+    impl<'a> Parser<'a> {
+        fn skip_whitespace(&mut self) {
+            while let Some(k) = self.o {
```

Comment:
This pattern seems to be used about 5 times in the code, so it's begging for a helper function receiving some sort of a predicate. I could also see `self.o` removed entirely if you used a peekable iterator. Actually, the `self.start` would go away as well ;) You'd only need to implement something like a `peek_while` for as a helper.